OneCommunityGlobal/HGNRest
# Kaikanes lessonlist likes routing/controller

# Description
Created a separate model LessonLike to handle keeping track of how many likes a Lesson has and which user liked it. And ability to unlike.

## Related PRS (if any):
https://github.com/OneCommunityGlobal/HGNRest/pull/654
https://github.com/OneCommunityGlobal/HGNRest/pull/635

## Main changes explained:
- created LikeSchema model
- added two property's to Lesson schema Likes and totalLikes
- created new put route for liking a Lesson 
-  added likeLesson function inside existing bmLessonController

## How to test:
1. check into current branch
2. do `npm install` ,  `npm run build` ,  `npm run dev` 
3. log into the app on the front end and grab your token in local storage
4. add 'Authorization' to the header with your token as the value
5. make sure 'content-type' header with 'application/json' value is in the header as well!
6. test put route for `http://localhost:4500/api/bm/lesson/:lessonId/like`
7. you will need to pass a `userId` in the body
8. grab a userId from `http://localhost:4500/api/bm/projects`
9. grab a lessonId from `http://localhost:4500/api/bm/lessons`
10. user should only be able to like a Lesson once. when they like it again it should unlike
11. `totalLikes` field should adjust accordingly
12. see video below




## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HGNRest/assets/114305309/23d6517b-3819-45ac-ac74-1a8c9ef784b9

GRAB TOKEN IN APP FRONT END LOCAL STORAGE
<img width="270" alt="image" src="https://github.com/OneCommunityGlobal/HGNRest/assets/114305309/dda410e3-9308-442c-898d-810e40dc5e92">
 HEADER (I use insomnia)
![image](https://github.com/OneCommunityGlobal/HGNRest/assets/114305309/0562e588-3448-4e6a-8cf2-9ceb6e372d04)

## Note:
The seed data/function will be deleted before merge
File names/variable/routing names might change so I can match other devs in progress Lesson post/get routes
